romancal/multiband_catalog/multiband_catalog_step.py

```diff
@@ -68,41 +68,71 @@ def process(self, library):
             example_model = library.borrow(0)
             library.shelve(example_model, modify=False)
 
-        source_catalog_model = datamodels.MultibandSourceCatalogModel.create_minimal(
+        # Initialize the source catalog model, copying the metadata
+        # from the input model
+        cat_model = datamodels.MultibandSourceCatalogModel.create_minimal(
             {"meta": example_model.meta}
         )
         if "instrument" in example_model.meta:
-            source_catalog_model.meta.optical_element = (
+            cat_model.meta.optical_element = (
                 example_model.meta.instrument.optical_element
             )
 
+        # Define the output filename for the source catalog model
         try:
-            source_catalog_model.meta.filename = library.asn["products"][0]["name"]
+            cat_model.meta.filename = library.asn["products"][0]["name"]
         except (AttributeError, KeyError):
-            source_catalog_model.meta.filename = "multiband_catalog"
-        if self.output_file is None:
-            self.output_file = source_catalog_model.meta.filename
+            cat_model.meta.filename = "multiband_catalog"
 
+        # Check that the input library does not contain a model with
+        # all invalid values
+        with library:
+            all_nans = []
+            for model in library:
+                mask = (
+                    ~np.isfinite(model.data)
+                    | ~np.isfinite(model.err)
+                    | (model.err <= 0)
+                )
+                if np.all(mask):
+                    all_nans.append(model.meta.filename)
+                library.shelve(model, modify=False)
+        if any(all_nans):
+            msg = f"All pixels in the following input models are masked: {', '.join(all_nans)}"
+            return save_empty_results(self, model.data.shape, cat_model, msg=msg)
```

Comment:
Thinking about this case... let's say I have 5 band imaging, and for some reason one of the bands has barely intersects / doesn't intersect the skycell and is empty, but the other bands look fine.  I think in that case the multiband catalog sees 4 normal images and one all NaN image.  The ideal behavior is probably to treat this as 5 band images but for all of the 5th band information to be placeholder.  Do you see that differently?

Reply:
@schlafly The issue was with the creation of the detection image.

I've updated this PR to robustly handle NaN propagation in the creation of the detection image.  With that change (and the removal of the initial short circuit), empty placeholder columns are now produced for filters with all NaN inputs.

This PR is ready for final review, thanks!